Source/WebCore/Modules/webauthn/AuthenticatorCoordinator.cpp

```diff
@@ -272,6 +307,16 @@ void AuthenticatorCoordinator::discoverFromExternalSource(const Document& docume
         ASSERT(!exception.message.isNull());
         promise.reject(exception.toException());
     };
+
+    if (m_isCancelling) {
+        m_queuedRequest = [weakThis = WeakPtr { *this }, weakFrame = WeakPtr { *frame }, clientDataJsonHash = WTFMove(clientDataJsonHash), requestOptions = WTFMove(requestOptions), scopeAndCrossOriginParent = WTFMove(scopeAndCrossOriginParent), callback = WTFMove(callback)]() mutable {
+            if (!weakThis || !weakFrame)
+                return;
+            const auto& options = requestOptions.publicKey.value();
+            weakThis->m_client->getAssertion(*weakFrame, weakFrame->document()->securityOrigin(), clientDataJsonHash, options, requestOptions.mediation, scopeAndCrossOriginParent, WTFMove(callback));
```

Comment:
Same comment about weakFrame.

Reply:
Also `document()->protectedSecurityOrigin()` to make sure we protect the security origin we pass.